frontend/public/components/monitoring/alert-manager-yaml-editor.tsx

```diff
@@ -28,70 +27,61 @@ const AlertManagerYAMLEditor: React.FC<AlertManagerYAMLEditorProps> = ({
 
   const [errorMsg, setErrorMsg] = React.useState(initErrorMsg);
   const [successMsg, setSuccessMsg] = React.useState();
-  const [inProgress, setInProgress] = React.useState(false);
-  const [alertManagerYamlStr, setAlertManagerYamlStr] = React.useState(
-    !_.isEmpty(encodedAlertManagerYaml) ? Base64.decode(encodedAlertManagerYaml) : '',
-  );
+  const alertManagerYamlStr = !_.isEmpty(encodedAlertManagerYaml)
+    ? Base64.decode(encodedAlertManagerYaml)
+    : '';
 
-  const save = (e) => {
-    e.preventDefault();
-    if (_.isEmpty(alertManagerYamlStr)) {
+  const save = (yaml: string) => {
+    if (_.isEmpty(yaml)) {
```

Comment:
We should really test that this parses as YAML before patching the config

Reply:
Is that with `safeLoad()`?  Or will `safeDump()` also throw error if invalid YAML?

Reply:
`safeLoad`. We should catch and display YAML parsing errors.

Reply:
Ok, updated `alert-manager-utils.patchAlertManagerConfig` to throw error if `safeLoad` fails.

Reply:
fixed